### PR TITLE
NATIVE-532: Build configs for metro

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **NATIVE**.

### CODE REVIEW

File: src/modules/common/renderHtml.js [hunk 1]
```diff
@@ -86,113 +87,115 @@ const renderJS = (files: PageResourceCacheStateType) => `
...
-  </style>
-</head>
-<body dir="${RTL_LANGUAGES.includes(language) ? 'rtl' : 'ltr'}">
-  <div id="measure-container">${html}</div>
-  <script>${renderJS(files)}</script>
-</body>
-</html>
-`
+const renderHtml = (html: string, files: PageResourceCacheStateType, theme: ThemeType, language: string) => {
+  return `
```
Comment (Schedulaar) [inline src/modules/common/renderHtml.js:91]: Was this change necessary? You might revert it (except for the font-change) to keep up the git history

File: build-configs/configs/BuildConfigType.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// @flow
+
+import type { ThemeType } from '../themes/ThemeType'
+import type { FeatureFlagsType } from './featureFlags'
+
+export type BuildConfigType = {|
+  appTitle: string,
+  cmsUrl: string,
+  switchCmsUrl?: string,
```
Comment (Schedulaar) [inline build-configs/configs/BuildConfigType.js:9]: maybe rather call this `testCmsUrl`

File: build-configs/configs/BuildConfigType.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// @flow
+
+import type { ThemeType } from '../themes/ThemeType'
+import type { FeatureFlagsType } from './featureFlags'
+
+export type BuildConfigType = {|
+  appTitle: string,
```
Comment (Schedulaar) [inline build-configs/configs/BuildConfigType.js:7]: Is it appTitle or appName? I feel we are inconsistent... Maybe lookup the change in locales.json

File: build-configs/configs/BuildConfigType.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  cmsUrl: string,
+  switchCmsUrl?: string,
+  featureFlags: FeatureFlagsType,
+  theme: ThemeType,
```
Comment (Schedulaar) [inline build-configs/configs/BuildConfigType.js:11]: I'd call it `brightTheme` in contrast to the darkTheme (no pun intended)

File: build-configs/themes/integreat/fonts.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import type { FontsType } from '../common/fonts'
+import { commonFonts } from '../common/fonts'
+
+const fonts: FontsType = {
+  decorativeFontBold: 'Raleway-Bold',
+  decorativeFontRegular: 'Raleway-Regular',
+  contentFontRegular: 'OpenSans-Regular',
+  contentFontBold: 'OpenSans-Bold',
+  arabicWebviewFontFamilies: '\'OpenSans\', \'Lateef\', sans-serif',
+  webviewFontFamilies: '\'OpenSans\', sans-serif',
```
Comment (Schedulaar) [inline build-configs/themes/integreat/fonts.js:12]: Afaics, these are also common, aren't they?

File: src/modules/theme/constants/webviewFontFamilies.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+import type { ThemeType } from './theme'
+
+const ARABIC_LANGUAGES = ['ar', 'fa', 'ku']
+
+const webviewFontFamilies = (theme: ThemeType, language: string) => {
+  return ARABIC_LANGUAGES.includes(language) ? theme.fonts.arabicWebviewFontFamilies : theme.fonts.webviewFontFamilies
+}
+
+export default webviewFontFamilies
```
Comment (Schedulaar) [inline src/modules/theme/constants/webviewFontFamilies.js:11]: I'd move this out of the constants folder.

File: src/modules/endpoint/sagas/loadCityContent.js [hunk 7]
```diff
@@ -93,7 +93,9 @@ export default function * loadCityContent (
   // loadLanguages did not update the dataContainer this is needed. In case the previous call to fetchResourceCache
   // failed to download some resources an other call could fix this and download missing files.
   if (criterion.shouldRefreshResources() && !isCellularConnection) {
-    const resourceURLFinder = new ResourceURLFinder(allowedResourceHostNames)
+    const { cmsUrl, switchCmsUrl } = buildConfig()
+    const allowedHostNames = switchCmsUrl ? [cmsUrl, switchCmsUrl] : [cmsUrl]
```
Comment (Schedulaar) [inline src/modules/endpoint/sagas/loadCityContent.js:97]: `cmsUrl` and `switchCmsUrl` are no hostnames!
Make sure `allowedHostNames` is the same as before for the integreat build.
Maybe we should just just add this attribute to the buildConfig.

File: src/modules/endpoint/determineApiUrl.js [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
 
 import AppSettings from '../settings/AppSettings'
-import { baseUrl } from './constants'
+import buildConfig from '../app/constants/buildConfig'
 
 export default async () => {
   const appSettings = new AppSettings()
   const apiUrlOverride = await appSettings.loadApiUrlOverride()
-  return apiUrlOverride || baseUrl
+  return apiUrlOverride || buildConfig().cmsUrl
```
Comment (Schedulaar) [inline src/modules/endpoint/determineApiUrl.js:9]: I guess, this should depend on `__DEV__` like 
```suggestion
  return apiUrlOverride || (__DEV__ ? buildConfig().testCmsUrl : buildConfig().cmsUrl)
```

File: src/modules/app/constants/buildConfig.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    throw Error('No BUILD_CONFIG_NAME supplied!')
+  }
+  const buildConfig: ?BuildConfigType = buildConfigs[buildConfigName]
+  if (!buildConfig) {
+    throw Error('Invalid BUILD_CONFIG_NAME supplied!')
+  }
+  return buildConfig
+}
+
+export default buildConfig
```
Comment (Schedulaar) [inline src/modules/app/constants/buildConfig.js:26]: Would be nice, if only the build config is included which we build for. I guess in the webapp it's solved better. Not sure, whether the technique of injecting the build-config is applicable here though.
I guess, we might just define a global __CONFIG__ object and let the bundler inject it here, but I am not sure, if that works.

File: src/.babelrc [hunk 10]
```diff
@@ -1,9 +0,0 @@
-{
-  "plugins": [
-    ["transform-inline-environment-variables", {
-      "include": [
-        "E2E_TEST_IDS"
-      ]
-    }]
-  ]
-}
```
Comment (Schedulaar) [inline src/.babelrc:9]: Is this no longer necessary?

File: src/routes/landing/components/EastereggImage.js [hunk 11]
```diff
@@ -43,14 +43,19 @@ class EastereggImage extends React.Component<PropsType, StateType> {
...
+
     const prevClickCount = this.state.clickCount
     const clickStart = this.state.clickStart
     const clickedInTimeInterval = clickStart && clickStart.isAfter(moment().subtract(CLICK_TIMEOUT, 's'))
 
     if (prevClickCount + 1 >= API_URL_OVERRIDE_MIN_CLICKS && clickedInTimeInterval) {
       const appSettings = new AppSettings()
       const apiUrlOverride = await appSettings.loadApiUrlOverride()
-      const newApiUrl = (apiUrlOverride === testBaseUrl) || (!apiUrlOverride && baseUrl === testBaseUrl) ? liveBaseUrl : testBaseUrl
+      const newApiUrl = !apiUrlOverride || apiUrlOverride === cmsUrl ? switchCmsUrl : cmsUrl
```
Comment (Schedulaar) [inline src/routes/landing/components/EastereggImage.js:58]:
```suggestion
      const newApiUrl = (!apiUrlOverride || apiUrlOverride === cmsUrl) ? switchCmsUrl : cmsUrl
```

File: build-configs/configs/BuildConfigType.js (hunk 2 above)
Comment (klinzo) [reply build-configs/configs/BuildConfigType.js:9]: Not really correct since for the integreat-test-cms config it is the other way round.

File: build-configs/configs/BuildConfigType.js (hunk 4 above)
Comment (klinzo) [reply build-configs/configs/BuildConfigType.js:11]: Maybe rather lightTheme?

File: build-configs/themes/integreat/fonts.js (hunk 5 above)
Comment (klinzo) [reply build-configs/themes/integreat/fonts.js:12]: Right now they are, but this will probably be changed with malte.

File: src/modules/endpoint/determineApiUrl.js (hunk 8 above)
Comment (klinzo) [reply src/modules/endpoint/determineApiUrl.js:9]: No, it should depend on the build config.

File: src/.babelrc (hunk 10 above)
Comment (klinzo) [reply src/.babelrc:9]: Moved to babel.config.json in the root.

File: src/modules/app/constants/buildConfig.js (hunk 9 above)
Comment (klinzo) [reply src/modules/app/constants/buildConfig.js:26]: Yes, we did kind of discuss this already. I'll probably create an issue to fix this as we have a little bit time pressure :D

File: build-configs/configs/BuildConfigType.js (hunk 4 above)
Comment (Schedulaar) [reply build-configs/configs/BuildConfigType.js:11]: Even better :)

File: src/modules/common/renderHtml.js (hunk 1 above)
Comment (klinzo) [reply src/modules/common/renderHtml.js]: No, I just assigned a constant with webviewFontFamilies at first but inlined that later. Should be fine now.

File: build-configs/configs/BuildConfigType.js (hunk 3 above)
Comment (klinzo) [reply build-configs/configs/BuildConfigType.js:7]: Just copied this from the webapp build config which is kind of old, but I guess you are right. Will change that.

File: src/modules/endpoint/determineApiUrl.js (hunk 8 above)
Comment (klinzo) [reply src/modules/endpoint/determineApiUrl.js:9]: At least that is my opinion. Otherwise we would not need a integreat-test-cms build config.

File: src/modules/endpoint/sagas/loadCityContent.js (hunk 7 above)
Comment (klinzo) [reply src/modules/endpoint/sagas/loadCityContent.js]: Added to build configs

File: .circleci/config.yml [hunk 12]
```diff
@@ -262,6 +287,9 @@ jobs:
       e2e_build:
         description: Whether to create a build for E2E testing
```
Comment (Schedulaar) [inline .circleci/config.yml:288]:
```suggestion
      e2e_build:
        description: Whether to include E2E test ids.
```

File: src/modules/endpoint/determineApiUrl.js (hunk 8 above)
Comment (Schedulaar) [reply src/modules/endpoint/determineApiUrl.js:9]: I'm fine with the current state. Doesn't really make a huge difference imo.

**Comment by klinzo** [comment]
@lulo4907 could you try again please?

File: .circleci/config.yml [hunk 13]
```diff
@@ -183,7 +191,10 @@ jobs:
   build_android:
     parameters:
       e2e_build:
-        description: Whether to create a build for E2E testing
+        description: Whether to create a build for E2E testing.
```
[the remark this answers is not in the document]
Comment (maxammann) [reply .circleci/config.yml:194]: I like it as it is because it is more generic. Test IDs will maybe not be the only thing needed in the future.

File: src/routes/settings/container/SettingsContainer.js [hunk 14]
```diff
@@ -22,5 +22,5 @@ const mapStateToProps = (state: StateType) => {
 }
 
 export default connect<PropsType, OwnPropsType, _, _, _, _>(mapStateToProps)(
-  withTheme(props => props.language)(withTranslation('settings')(Settings))
+  withTheme(withTranslation('settings')(Settings))
```
Comment (maxammann) [inline src/routes/settings/container/SettingsContainer.js:25]: You can remove the connect from this component. `language` and `dispatch` is unused in the Settings component

File: src/modules/theme/hocs/__mocks__/withTheme.js [hunk 15]
```diff
@@ -1,25 +0,0 @@
-// @flow
```
[the remark this answers is not in the document]
Comment (klinzo) [reply src/modules/theme/hocs/__mocks__/withTheme.js:1]: Because the mock was exactly the same as the hoc is now.

File: src/routes/settings/container/SettingsContainer.js (hunk 14 above)
Comment (klinzo) [reply src/routes/settings/container/SettingsContainer.js:25]: Nope! :D
